workerpool/pool.go

```diff
@@ -38,23 +38,33 @@ func New(session *concurrency.Session, maxWorkers uint, events <-chan *watcher.E
 
 // Start listens on events channel and tries to start a job for every connection.
 func (pool *WorkerPool) Start() {
-	for {
-		event, more := <-pool.events
-		if more {
-			if event.Type == watcher.Created {
-				job, err := newJob(pool.session, event.Connection, pool.log.Named("job"))
-				if err != nil {
-					pool.log.Debugw("creating new job failed", "error", err, "connectionID", event.ID)
-					continue
+	go func() {
+		for {
+			event, more := <-pool.events
+			if more {
+				pool.log.Debugw("event received", "connectionID", event.ID, "type", event.Type)
+
+				switch event.Type {
+				case watcher.Created:
+					job, err := newJob(pool.session, event.Connection, pool.log.Named("job"))
+					if err != nil {
+						pool.log.Debugw("creating new job failed", "error", err, "connectionID", event.ID)
+						continue
+					}
+
+					job.start()
+
+					pool.jobs[event.ID] = job
+					pool.numWorkers += event.Connection.Source.NumberOfWorkers()
+				case watcher.Deleted:
+					if job, exists := pool.jobs[event.ID]; exists {
+						job.stop()
+						delete(pool.jobs, event.ID)
+					}
```

Comment:
Should also decrement the `pool.numWorkers` by the number we drop.

Reply:
@sebito91 done.